applications/StructuralMechanicsApplication/custom_elements/updated_lagrangian.cpp

```diff
@@ -0,0 +1,565 @@
+                rThisKinematicVariables.F(2, index) = 0.0;
+            }
+
+            rThisKinematicVariables.N = GetGeometry().ShapeFunctionsValues( rThisKinematicVariables.N, IntegrationPoints[PointNumber].Coordinates() );
+            const double current_radius = StructuralMechanicsMathUtilities::CalculateRadius(rThisKinematicVariables.N, GetGeometry(), Current);
+            const double initial_radius = StructuralMechanicsMathUtilities::CalculateRadius(rThisKinematicVariables.N, GetGeometry(), Initial);
+            rThisKinematicVariables.F(2, 2) = current_radius/initial_radius;
+        }
+        
+        // Here we essentially set the input parameters
+        rThisKinematicVariables.detF = MathUtils<double>::Det(rThisKinematicVariables.F);
+        const double detFT = rThisKinematicVariables.detF * mDetF0[PointNumber];
+        rValues.SetDeterminantF(detFT); //assuming the determinant is computed somewhere else
+        const Matrix FT = prod(rThisKinematicVariables.F, mF0[PointNumber]);
+        rValues.SetDeformationGradientF(FT); //F computed somewhere else
+        
+        // Here we set the space on which the results shall be written
+        rValues.SetConstitutiveMatrix(rThisConstitutiveVariables.D); //assuming the determinant is computed somewhere else
+        rValues.SetStressVector(rThisConstitutiveVariables.StressVector); //F computed somewhere else
+        
+        // Actually do the computations in the ConstitutiveLaw    
+        mConstitutiveLawVector[PointNumber]->CalculateMaterialResponse(rValues, ThisStressMeasure); //here the calculations are actually done 
+    }
+    
+    //************************************************************************************
+    //************************************************************************************
+    
+    double UpdatedLagrangian::CalculateDerivativesOnReference(
+        Matrix& J0, 
+        Matrix& InvJ0, 
+        Matrix& DN_DX, 
+        const unsigned int PointNumber,
+        IntegrationMethod ThisIntegrationMethod
+        )
+    {
+        double detJ0;
+        
+        const Matrix& DN_De = GetGeometry().ShapeFunctionsLocalGradients(ThisIntegrationMethod)[PointNumber];
+        
+        J0 = GetGeometry().Jacobian( J0, PointNumber, ThisIntegrationMethod);
+        
+        MathUtils<double>::InvertMatrix( J0, InvJ0, detJ0 );
+        
+        noalias( DN_DX ) = prod( DN_De, InvJ0);
+        
+        return detJ0;
+    }
+    
+    //************************************************************************************
+    //************************************************************************************
+    
+    double UpdatedLagrangian::CalculateDerivativesOnCurrent(
+        Matrix& J, 
+        Matrix& InvJ, 
+        Matrix& DN_DX, 
+        const unsigned int PointNumber,
+        IntegrationMethod ThisIntegrationMethod
+        )
+    {
+        J.clear();
+        
```

Comment:
I do not follow.

is this different from what done in the base class? (and if so, why did u implement at all this funciton in the base class???)

most importantly, please document very clearly at the beginning of the file what you are considering as reference configuration